### ISSUE TITLE
Some thinking in getter

### ISSUE THREAD

**Issue by leaves4j** [vuejs/vuex#207]
like this:

    {
      vuex: {
        getter: {
          local_data: state => state.store_data
          }
      }
    }

I think it's fine that `local_data` changing when `state.store_data` changed, but `local_data` can changed by local component, and doesn't effect `state.store_data` . this idea also like this:

    {
      vuex: {
        getter: {
          store_data: state => state.store_data
        }
      },
      watch {
        store_data() {
          this.local_data = this.store_data;
        }
      }
    }
In this way, it's better in form handling,we can use `v-model` and also can resolve some thrid part component doesn't have the 'change'  event

**Comment by fnlctrl** [comment]
If you want vuex to manage your states, you should't mutate those states locally. Instead, use store `actions` to mutate the states inside store itself.

**Comment by leaves4j** [comment]
@fnlctrl Thank for your answer,Maybe you didn't understand what I mean, this doesn't matter data in loacl or vuex stores, just a idea for the getter's behavior, and I don't think vuex should manage all states, some temporary data,changing form data is better deal in local,The vuex store final results

**Comment by fnlctrl** [comment]
Yes, it would be very verbose if vuex should manage all states, but why would you want to let vuex store the final results? If you need to submit the final results, why not just do it inside the component?

**Comment by jonagoldman** [comment]
@leaves4j you issue is not clear.
Are you saying that `local_data` can be modified in component by doing `this.local_data == 'something'` and this will not affect vuex state itself?

**Comment by leaves4j** [comment]
yes


原始邮件
发件人:Jona Goldmannotifications@github.com
收件人:vuejs/vuexvuex@noreply.github.com
抄送:jiangqjiangq.leaves+github@gmail.com; Mentionmention@noreply.github.com
发送时间:2016年6月13日(周一) 17:52
主题:Re: [vuejs/vuex] Some thinking in getter (#207)


@leaves4j you issue is not clear.
 Are you saying that local_data can be modified in component by doing this.local_data == 'something' and this will not affect vuex state itself?

**Comment by leaves4j** [comment]
Maybe I didn't describe clearly, look at the following three sections of the code:
a component with `change` event, and `change` event will set the changed value as parameter

a component with `change` event, and `change` event will set the changed value as parameter

```js
<template>
  <div>
    <input :value="message" @input="updateMessage">
    <button @click="saveMessage"></button>
  </div>
</template>
<style scoped lang="scss">
</style>
<script>

  export default{
    data(){
      return {}
    },
    vuex: {
      getters: {
        message: state => state.obj.message
      },
      actions: {
        updateMessage: ({ dispatch }, e) => {
          dispatch('UPDATE_MESSAGE', e.target.value);
        },
        saveMessage: ({ dispatch }) => {
          dispatch('SAVE_MESSAGE');
        }
      }
    }
  }
</script>

```

a component without `change` event

```js
<template>
  <div>
    <some-component :value.sync="local_message"></some-component>
    <button @click="save"></button>
  </div>
</template>
<style scoped lang="scss">
</style>
<script>
I
  export default{
    data(){
      return {
        local_message: 'init_value'
      }
    },
    vuex: {
      getters: {
        message: state => state.obj.message
      },
      actions: {
        saveMessage: ({dispatch}, value) => {
          dispatch('SAVE_MESSAGE', value)
        }
      }
    },
    watch: {
      message(){
        this.local_message = this.message;
      }
    },
    methods: {
      save(){
        this.saveMessage(this.local_message);
      }
    }
  }
</script>
```

in my idea

```js
<template>
  <div>
    <some-component :value.sync="message"></some-component>
    <button @click="save"></button>
  </div>
</template>
<style scoped lang="scss">
</style>
<script type="text/ecmascript-6">

  export default{
    data(){
      return {}
    },
    vuex: {
      getters: {
        message: state => state.obj.message
      },
      actions: {
        saveMessage: ({dispatch}, value) => {
          dispatch('SAVE_MESSAGE', value)
        }
      }
    },
    methods: {
      save(){
        this.saveMessage(this.message);
      }
    }
  }
</script>
```

**Comment by fnlctrl** [comment]
After going through your code I don't think it's a good idea trying to omit `watch`-es and let vue magically 'watch' the value using a 'getter'. It's a disaster.
I never thought it would be any good using vuex in form handling... What's the point of recording every mutation of user input? Since you chose to use vuex for form handling, the cost of being super verbose is inevitable.

**Comment by fnlctrl** [comment]
You can also use a computed property for this, checkout http://vuex.vuejs.org/en/forms.html
> Yet an alternative approach to leverage v-model ...

**Comment by fnlctrl** [comment]
You can also use a computed property for this, checkout http://vuex.vuejs.org/en/forms.html
> Yet an alternative approach to leverage v-model ...

**Comment by leaves4j** [comment]
I simplified the problem by data flow
normal v-model:
![image](https://cloud.githubusercontent.com/assets/6735681/16006978/75f99fbc-31a2-11e6-841c-a50811d8e64c.png)
normal vuex data flow
![image](https://cloud.githubusercontent.com/assets/6735681/16006988/90243028-31a2-11e6-82cb-5bb6553b845d.png)
what i think
![image](https://cloud.githubusercontent.com/assets/6735681/16006994/99d0d7de-31a2-11e6-8a07-cb6954af6b15.png)
